File: src/main/java/hilfling/backend/hilfling/controllers/EventOwnerController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.EventOwner;
import hilfling.backend.hilfling.service.EventOwnerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/event_owners")
public class EventOwnerController extends GenericBaseControllerImplementation<EventOwner> {
    @Autowired
    private EventOwnerService service;
    public EventOwnerService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/EventOwnerController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.*
import hilfling.backend.hilfling.repository.AlbumRepository
import hilfling.backend.hilfling.repository.ArticleRepository
import hilfling.backend.hilfling.repository.CategoryRepository
import hilfling.backend.hilfling.repository.EventOwnerRepository
import me.liuwj.ktorm.database.Database
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/event_owners")
class EventOwnerController {
    val repository = EventOwnerRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Long): EventOwner? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll(): List<EventOwner> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("name") name: String
    ): EventOwner {
        return repository.create(
                name
        )
    }
}